diff --git a/libavformat/file.c b/libavformat/file.c
index 8303436be09..b3d19954fb4 100644
--- a/libavformat/file.c
+++ b/libavformat/file.c
@@ -395,7 +395,9 @@ static int pipe_open(URLContext *h, const char *filename, int flags)
     setmode(fd, O_BINARY);
 #endif
     c->fd = fd;
-    h->is_streamed = 1;
+
+    struct stat st;
+    h->is_streamed = !fstat(fd, &st) && S_ISFIFO(st.st_mode);
     return 0;
 }
 
@@ -404,6 +406,7 @@ const URLProtocol ff_pipe_protocol = {
     .url_open            = pipe_open,
     .url_read            = file_read,
     .url_write           = file_write,
+    .url_seek            = file_seek,
     .url_get_file_handle = file_get_handle,
     .url_check           = file_check,
     .priv_data_size      = sizeof(FileContext),
